src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -168,6 +169,73 @@ class StatementSegment(ansi_dialect.get_segment("StatementSegment")):  # type: i
     )
 
 
+@snowflake_dialect.segment()
+class ClusterBySegment(BaseSegment):
+    """A `CLUSTER BY` segment."""
+
+    type = "clusterby_segment"
+    # https://docs.snowflake.com/en/user-guide/tables-clustering-keys.html#defining-a-clustering-key-for-a-table
+
+    match_grammar = StartsWith(
+        "CLUSTER",
+        "BY",
+        Bracketed(
+            Delimited(
+                Sequence(
+                    OneOf(
+                        # Can CLUSTER BY a column
+                        Ref("ColumnReferenceSegment"),
+                        # Can CLUSTER BY an expression (e.g. `TO_DATE(timestamp_column)`)
+                        Ref("ExpressionSegment"),
+                    ),
+                ),
+            ),
+        ),
+    )
+
+
+@snowflake_dialect.segment(replace=True)
+class CreateTableStatementSegment(BaseSegment):
+    """A `CREATE TABLE` statement."""
+
+    type = "create_table_statement"
+    # https://docs.snowflake.com/en/sql-reference/sql/create-table.html
+
+    match_grammar = Sequence(
+        "CREATE",
+        Ref("OrReplaceGrammar", optional=True),
+        Ref("TemporaryTransientGrammar", optional=True),
+        "TABLE",
+        Ref("IfNotExistsGrammar", optional=True),
+        Ref("TableReferenceSegment"),
+        OneOf(
+            # Columns and comment syntax:
+            Sequence(
+                Bracketed(
+                    Delimited(
+                        OneOf(
+                            Ref("TableConstraintSegment"),
+                            Ref("ColumnDefinitionSegment"),
+                        ),
+                    ),
+                ),
+                OneOf(
+                    Ref("CreateStatementCommentSegment", optional=True),
+                    Ref("ClusterBySegment", optional=True),
+                    optional=True,
+                ),
+            ),
+            # Create AS syntax:
+            Sequence(
+                "AS",
+                OptionallyBracketed(Ref("SelectableGrammar")),
+            ),
+            # Create like syntax
+            Sequence("LIKE", Ref("TableReferenceSegment")),
+        ),
+    )
```

Comment:
This duplicates a lot of code from the original `CreateTableStatementSegment`, and we've had a few bugs recently where duplicate code like this has got out of sync. What I'd prefer to do is to define a new grammar or segment, which defines the following in the original `ansi` dialect:

```
Sequence(  # [COMMENT 'string'] (MySQL)
    "COMMENT", Ref("QuotedLiteralSegment"), optional=True
),
```

Then, instead of redifining the whole of `CreateTableStatementSegment` in the snowflake dialect we only need to redefine that grammar or segment as:

```
OneOf(
    Ref("CreateStatementCommentSegment", optional=True),
    Ref("ClusterBySegment", optional=True),
),
```

in the snowflake dialect.

Reply:
Splitting the refactoring work in a separate commit: https://github.com/sqlfluff/sqlfluff/pull/923 - I'll update this PR once the other's merged. 🙏